frontend/integration-tests/tests/add-secrets.scenario.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable max-nested-callbacks */
+import { appHost, testName, checkLogs, checkErrors } from '../protractor.conf';
+import { execSync } from 'child_process';
+import * as secretsView from '../views/secrets.view';
+import * as addSecretView from '../views/modal-add-secret-to-workload.view';
+import { searchJSON } from '../views/modal-add-secret-to-workload.view';
+
+describe('Add Secret to Workloads', () => {
+  const secretName = 'test-secret';
+  const resourceName = 'example';
+  const resourceKind = 'deployment';
+  const envPrefix = 'env-';
+  const mountPath = '/tmp/testdata';
+
+
+  beforeAll(async()=> {
+    // create deployment and secret
+    execSync(`kubectl run ${resourceName} --image=aosqe/hello-openshift -n ${testName}`);
+    execSync(`kubectl create secret generic ${secretName} --from-literal=key1=supersecret -n ${testName}`);
+  });
+
+  beforeEach(async() => secretsView.visitOneSecretPage(appHost, testName, secretName));
+
+  afterEach(() => {
+    checkLogs();
+    checkErrors();
+  });
+
+  describe('Add Secret to Workloads as Enviroment Variables', ()=> {
+    it('Add Secret to Deployment as Env', async() => {
+      await addSecretView.addSecretToWorkloadAsEnv(resourceName, envPrefix);
+      //console.log("=== printing result");
+      //console.log("=== printing result");
+      await new Promise(resolve => (function checkForPackages() {
+        const output = execSync(`kubectl get -o json -n ${testName} ${resourceKind} ${resourceName}`).toString('utf-8');
+        if (JSON.parse(output).status.observedGeneration === 2 ) {
+          return resolve();
+        }
+        setTimeout(checkForPackages, 2000);
+      })());
+      expect(searchJSON('spec.template.spec.containers[0].envFrom[0].secretRef.name', secretName, resourceName, testName, resourceKind)).toBe(true);
+      expect(searchJSON('spec.template.spec.containers[0].envFrom[0].prefix', envPrefix, resourceName, testName, resourceKind)).toBe(true);
+    });
+  });
+
+  describe('Add Secret to Workloads as Volume', ()=> {
+    it('Add Secret to Deployment as Vol', async() => {
+      await addSecretView.addSecretToWorkloadAsVol(resourceName, mountPath);
+      await new Promise(resolve => (function checkForPackages() {
+        const output = execSync(`kubectl get -o json -n ${testName} ${resourceKind} ${resourceName}`).toString('utf-8');
```

Comment:
Here we should use the create `getResourceJSON()`

Reply:
you're right, I will update to use `getResourceJSON()`